R/sortablestr2i.R

```diff
@@ -0,0 +1,42 @@
+# Alternative to hashstr2i that can produce semi-sorted chunks
+# Apply as e.g.:
+# split_values <- map(dff, sample_n, size=1) %>% 
+#   select(c("id1", "id2")) %>%
+#   collect() %>%
+#   arrange(id1, id2)
+# 
+# shard_by_rule <- splitstr2i(split_values)
+# code = glue::glue("df[,.out.disk.frame.id := {shard_by_rule}]")
+
+sortablestr2i <- function(split_values, desc_vars){
+  escape <- function(x) {
+    if(is.character(x)){
+      paste0("\"", x, "\"")
+    } else {
+      x
+    }
+  }
+  
+  switchcond <- function(name){ 
+      paste0("(",
+             name, 
+             ifelse(name %in% desc_vars, " < ", " > "),
+             escape(split_values[,name]), 
+             " | (",
+             name, 
+             " == ",
+             escape(split_values[,name])
+             )
+  }
+  
+  do.call(
```

Comment:
add comments about what this is doing and examples of output. I think we should separate this out as a function for generating the `if-else` and have some tests cases.

Reply:
I mean the `switchcond` function and the below

Reply:
Function separated, comments added. Not sure how to best set up the test cases.